projects/igniteui-angular/src/lib/calendar/months-view/months-view.component.ts

```diff
@@ -259,11 +259,13 @@ export class IgxMonthsViewComponent implements ControlValueAccessor {
         }
 
         const months = this.dates.toArray();
-        if (months.indexOf(node) - 3 >= 0) {
-            const month = months[months.indexOf(node) - 3];
-
-            month.nativeElement.focus();
-         }
+        for (let index = months.indexOf(node) - 1; index >= 0; index--) {
+            if (node.nativeElement.getBoundingClientRect().top !== months[index].nativeElement.getBoundingClientRect().top &&
```

Comment:
`node.nativeElement.getBoundingClientRect()` could be extracted as a variable outside of the cycle as it doesn't depend on the `index`.